android/src/main/kotlin/com.netguru.multiplatformstorage/AndroidStorageProvider.kt

```diff
@@ -25,19 +52,123 @@ internal object AndroidStorageProvider {
      * In other cases, it uses default instance.
      */
     internal fun preferences(name: String? = null): Preferences {
+        // Generate a new encryption key with initialization vector.
+        val keyStore = KeyStore.getInstance(KEYSTORE_NAME)
+        keyStore.load(null)
+
+        val preferences = appContext.getSharedPreferences(PREFERENCES_NAME, Context.MODE_PRIVATE)
+
+        // Generating new keys
+        if (!keysExist(keyStore, preferences)) {
+            clearKeys(keyStore, preferences)
+            generateKeystoreKey(KEYSTORE_VALUES_KEY_ALIAS, keyStore)
+            generateKeystoreKey(KEYSTORE_VALUES_IV_ALIAS, keyStore)
+            generateKeystoreKey(KEYSTORE_KEYS_KEY_ALIAS, keyStore)
+            preferences.edit {
+                putString(VALUES_KEY_ALIAS, generateEncryptedKey(keyStore.getSecretKey(KEYSTORE_VALUES_KEY_ALIAS)))
+                putString(VALUES_IV_ALIAS, generateEncryptedKey(keyStore.getSecretKey(KEYSTORE_VALUES_IV_ALIAS)))
+                putString(KEYS_KEY_ALIAS, generateEncryptedKey(keyStore.getSecretKey(KEYSTORE_KEYS_KEY_ALIAS)))
+            }
+        }
+
+        val valuesKey = getDecryptedKey(VALUES_KEY_ALIAS, KEYSTORE_VALUES_KEY_ALIAS, preferences, keyStore)
+        val valuesIv = getDecryptedKey(VALUES_IV_ALIAS, KEYSTORE_VALUES_IV_ALIAS, preferences, keyStore)
+        val keysKey = getDecryptedKey(KEYS_KEY_ALIAS, KEYSTORE_KEYS_KEY_ALIAS, preferences, keyStore)
```

Comment:
kaysKey? maybe some more meaningfull? `MasterKey` or smth?

Reply:
IMO this one fits better because this key is used to encrypt keys in key-value pairs